### PR TITLE
FIX

### PR DESCRIPTION
**Pull request by FrancoisGe**

### CODE REVIEW

**Comment by kebeclibre** [comment]
```
diff --git a/addons/web/static/src/legacy/legacy_views.js b/addons/web/static/src/legacy/legacy_views.js
index e2bd9c1294d..21e810de946 100644
--- a/addons/web/static/src/legacy/legacy_views.js
+++ b/addons/web/static/src/legacy/legacy_views.js
@@ -101,10 +101,20 @@ function registerView(name, LegacyView) {
                     viewFields: result.fields_views.search.fields,
                 });
             }
+            const viewSwitcherEntries = this.props.viewSwitcherEntries.slice();
+            this.viewParams.action.views.forEach(([vid, vtype]) => {
+                if (!viewSwitcherEntries.find((v) => v.type === vtype)) {
+                    viewSwitcherEntries.push({
+                        viewID: vid,
+                        type: vtype,
+                        multiRecord: !this.constructor.multiRecord,
+                    });
+                }
+            });
             this.viewParams.action = Object.assign({}, this.viewParams.action, {
                 controlPanelFieldsView,
                 _views: this.viewParams.action.views,
-                views: this.props.viewSwitcherEntries,
+                views: viewSwitcherEntries,
             });
         }
     }

```

File: addons/web/static/src/legacy/legacy_views.js [hunk 1]
```diff
@@ -101,10 +101,23 @@ function registerView(name, LegacyView) {
                     viewFields: result.fields_views.search.fields,
                 });
             }
+            const views = this.props.viewSwitcherEntries.slice();
+            this.viewParams.action.views.forEach(([vid, vtype]) => {
+                const view = views.find((v) => v.type === vtype);
+                if (view) {
+                    Object.assign(view, {viewID: vid});
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/legacy_views.js]: Object.assign({}, view, {viewID: vid});

File: addons/web/static/tests/legacy/views/legacy_views_tests.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+/** @odoo-module */
+
+import { createWebClient, doAction } from "@web/../tests/webclient/helpers";
+import { patchWithCleanup } from "@web/../tests/helpers/utils";
+
+import ListView from 'web.ListView'
+
+QUnit.module("LegacyViews", () => {
```
Comment (aab-odoo) [inline addons/web/static/tests/legacy/views/legacy_views_tests.js]: i think you can move this to legacy_tests.js (in  webclient/actions)

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**